mod/consensus/pkg/cometbft/service/abci.go

```diff
@@ -90,75 +89,76 @@ func (s *Service[LoggerT]) InitChain(
 		// InitChain represents the state of the application BEFORE the first
 		// block, i.e. the genesis block. This means that when processing the
 		// app's InitChain handler, the block height is zero by default.
-		// However, after Commit is called
-		// the height needs to reflect the true block height.
-		initHeader.Height = req.InitialHeight
+		// However, after genesis is handled the height needs to reflect
+		// the true block height.
+		initHeader := cmtproto.Header{
+			ChainID: req.ChainId,
+			Time:    req.Time,
+			Height:  req.InitialHeight,
+		}
 		s.finalizeBlockState.SetContext(
 			s.finalizeBlockState.Context().WithBlockHeader(initHeader),
 		)
 	}()
 
-	if s.finalizeBlockState == nil {
-		return nil, errors.New("finalizeBlockState is nil")
-	}
-
-	res, err := s.initChainer(s.finalizeBlockState.Context(), req)
+	resValidators, err := s.initChainer(
+		s.finalizeBlockState.Context(),
+		req.AppStateBytes,
+	)
 	if err != nil {
 		return nil, err
 	}
 
-	if res == nil {
+	if len(resValidators) == 0 {
 		return nil, errors.New(
-			"application init chain handler returned a nil response",
+			"application init chain handler returned a no validators",
 		)
 	}
```

Comment:
this is an hollow check, since we generate res in `initChainer` (it's not the result of `s.Middleware.InitGenesis`).
Replaced with what I understand it's the check we want